### PR TITLE
[REVIEW] Ports Modular Computers from Baystation

### PR DESCRIPTION
**Pull request by Novacat**
Mostly Feature Complete. Needs extensive refinement and testing. So far, seems to work fine.

Modular computers are basically a replacement of old Computer3 code, which is currently running on laptops. They are capable of running a wide variety of programs, and have three form factors, from stationary consoles, to bulky but portable laptops, to completely portable tablets. The system is entirely modular, built off of NanoUI, and can be expanded in the future.

Also, a bunch of other miscellaneous changes got packed in, due to being tightly interwoven with the modular computer framework.

While Baystation went the route of completely replacing their stationary consoles with modular computers, I have not quite gone that far, trying to keep the two systems functioning side by side.

Known Issues:

### CODE REVIEW

File: code/modules/modular_computers/file_system/programs/camera.dm [hunk 1]
```diff
@@ -6,15 +6,15 @@
 	switch(network)
 		if(NETWORK_THUNDER)
 			return 0
-		if(NETWORK_ENGINE || NETWORK_ENGINEERING || NETWORK_ENGINEERING_OUTPOST || NETWORK_ALARM_ATMOS || NETWORK_ALARM_FIRE || NETWORK_ALARM_POWER)
+		if(NETWORK_ENGINE,NETWORK_ENGINEERING,NETWORK_ENGINEERING_OUTPOST,NETWORK_ALARM_ATMOS,NETWORK_ALARM_FIRE,NETWORK_ALARM_POWER)
```
Comment (Atermonera) [inline code/modules/modular_computers/file_system/programs/camera.dm]: Not sure `if()` statements work that way?

Comment (Novacat) [reply code/modules/modular_computers/file_system/programs/camera.dm]: Yeah, that one is odd, but it was deliberately done in https://github.com/Baystation12/Baystation12/pull/14227

I am guessing something weird was going on there.

Comment (Atermonera) [reply code/modules/modular_computers/file_system/programs/camera.dm]: Ooooh, it's switch syntax. How you specify multiple discrete cases, per byond documentation:
docs

**Comment by Novacat** [comment]
Mostly feature complete. All that is left is testing, fixing up bugs, that kind of things.

**Comment by Neerti** [comment]
At the time of posting, the server is now running this branch, so that it can be tested in a real environment.

**Comment by Neerti** [comment]
The SPACE magazine might need to get cut out since it looks like its Bay's lore and not ours.

**Comment by PrismaticGynoid** [comment]
There doesn't seem to be any programs for accessing records, which I feel is a pretty important thing to have.

**Comment by Novacat** [comment]
Bay's recordkeeping program is also paired with an extensive overhaul of how crew records are stored and accessed. I did not want to bloat this PR more than it was already.

As for the space magazine, I largely kept the first one just as an example, though, I can probably comment it out so that it cannot be accessed ingame.

**Comment by Neerti** [comment]
Trying to use the suit sensor program at CentCom resulted in the runtime below.
```
[2019-04-27T16:13:44] Runtime in suit_sensors.dm,54: list index out of bounds
   proc name: ui interact (/datum/nano_module/crew_monitor/ui_interact)
   usr: (the floor) (91,149,8) (/turf/unsimulated/floor) (neerti)
   usr.loc: (the floor) (91,149,8) (/turf/unsimulated/floor) (neerti)
   src: Crew monitor (/datum/nano_module/crew_monitor)
   call stack:
   Crew monitor (/datum/nano_module/crew_monitor): ui interact(Ellis Singh (/mob/observer/dead), "main", /datum/nanoui (/datum/nanoui), 1, /datum/topic_state/default (/datum/topic_state/default))
   /datum/computer_file/program/s... (/datum/computer_file/program/suit_sensors): ui interact(Ellis Singh (/mob/observer/dead), "main", null, 1)
   the console (/obj/item/modular_computer/console/preset/ert): ui interact(Ellis Singh (/mob/observer/dead), "main", /datum/nanoui (/datum/nanoui), 0, null, /datum/topic_state/default (/datum/topic_state/default))
   /datum/nanoui (/datum/nanoui): update(0)
   /datum/nanoui (/datum/nanoui): process(1)
   NanoUI (/datum/controller/subsystem/nanoui): update uis(the console (/obj/item/modular_computer/console/preset/ert))
   /datum/nanoui (/datum/nanoui): Topic("src=\[0x2105caa4];PC_runprogra...", /list (/list))
   Neerti (/client): Topic("src=\[0x2105caa4];PC_runprogra...", /list (/list), /datum/nanoui (/datum/nanoui))
```

**Comment by Neerti** [comment]
Another runtime, about a visible message.
```
[2019-04-27T16:29:47] Runtime in ,: bad arg name 'range'
   proc name: visible message (/atom/proc/visible_message)
   src: the laptop computer (/obj/item/modular_computer/laptop)
   src.loc: the plating (153,76,2) (/turf/simulated/floor/plating)
   call stack:
   the laptop computer (/obj/item/modular_computer/laptop): visible message("<span class=\'warning\'>The la...", null)
   /datum/computer_file/program/a... (/datum/computer_file/program/access_decrypter): event networkfailure(0)
   the laptop computer (/obj/item/modular_computer/laptop): process(20)
   Objects (/datum/controller/subsystem/processing/obj): fire(0)
   Objects (/datum/controller/subsystem/processing/obj): ignite(0)
   Master (/datum/controller/master): RunQueue()
   Master (/datum/controller/master): Loop()
   Master (/datum/controller/master): StartProcessing(0)
```

**Comment by Neerti** [comment]
I think you meant to block ``ID Computer`` instead of just the hop's office in the decryption program, since currently you can hack that access and go visit the desk in arrivals for all the access, which seems unintended considering HoP Office is blocked.
Decrypter UI

**Comment by Novacat** [comment]
The Centcom program runtime is probably caused by a bug with the suit sensors nanomodule itself, which is largely unchanged by this PR.

**Pull request merged**